extensions/parquet/base/src/main/java/io/deephaven/parquet/base/ColumnChunkReaderImpl.java

```diff
@@ -47,19 +46,17 @@ public class ColumnChunkReaderImpl implements ColumnChunkReader {
 
     ColumnChunkReaderImpl(
             ColumnChunk columnChunk, SeekableChannelsProvider channelsProvider,
-            Path rootPath, ThreadLocal<CodecFactory> codecFactory, MessageType type,
+            Path rootPath, DeephavenCodecFactory codecFactory, MessageType type,
             OffsetIndex offsetIndex, List<Type> fieldTypes) {
         this.channelsProvider = channelsProvider;
         this.columnChunk = columnChunk;
         this.rootPath = rootPath;
         this.path = type
                 .getColumnDescription(columnChunk.meta_data.getPath_in_schema().toArray(new String[0]));
         if (columnChunk.getMeta_data().isSetCodec()) {
-            decompressor = ThreadLocal.withInitial(() -> codecFactory.get().getDecompressor(
-                    CompressionCodecName.valueOf(columnChunk.getMeta_data().getCodec().name())));
+            decompressor = codecFactory.getByName(columnChunk.getMeta_data().getCodec().name());
         } else {
-            decompressor = ThreadLocal.withInitial(
-                    () -> codecFactory.get().getDecompressor(CompressionCodecName.UNCOMPRESSED));
+            decompressor = Compressor.PASSTHRU;
```

Comment:
Any reason we don't just get this for the `null` codec name to simplify matters?

Reply:
If you say "I have a codec, but the name is nonsense/null" I think we want to fail (as in your other comment), rather than assume that the data is just uncompressed, since we explicitly know that the compression codec flag was set.